src/upload-button.tsx

```diff
@@ -0,0 +1,384 @@
+            const inputFile = event.target.files[i];
+            console.log(inputFile, resolution);
+
+            if (resolution && applyToAll && resolution.skip)
+                break;
+
+            if (resolution && applyToAll && resolution.replace)
+                toUploadFiles.push(inputFile);
+            else {
+                let uploadFile = true;
+                for (const file of files) {
+                    if (file.name === inputFile.name) {
+                        const waiter = new Waiter();
+                        Dialogs.show(
+                            <FileConflictDialog
+                              path={path}
+                              file={file}
+                              uploadFile={inputFile}
+                              waiter={waiter}
+                              showApplyAll={event.target.files.length > 1}
+                            />
+                        );
+
+                        try {
+                            resolution = await waiter.block();
+                        } catch (exc) {
+                            resetInput();
+                            return;
+                        }
+
+                        if (resolution && resolution.skip)
+                            uploadFile = false;
+
+                        if (resolution.applyToAll)
+                            applyToAll = true;
+
+                        break;
+                    }
+                }
+
+                if (uploadFile)
+                    toUploadFiles.push(inputFile);
+            }
+        }
+
+        if (toUploadFiles.length === 0) {
+            resetInput();
+            return;
+        }
+
+        window.addEventListener("beforeunload", beforeUnloadHandler);
+
+        const cancelledUploads = [];
+        await Promise.allSettled(toUploadFiles.map(async (file: File) => {
+            const tmp_path = path.slice();
+            tmp_path.push(file.name);
+            const destination = tmp_path.join('/');
+            const abort = new AbortController();
+
+            setUploadedFiles(oldFiles => {
+                return {
+                    [file.name]: { file, progress: 0, cancel: () => abort.abort() },
+                    ...oldFiles,
+                };
+            });
+
+            try {
+                await upload(destination, file, (progress) => {
+                    // FIXME: this is an awful hack
+                    const now = performance.now();
+                    if (now < next_progress)
+                        return;
+                    next_progress = now + 200; // only rerender every 200ms
+                    setUploadedFiles(oldFiles => {
+                        const oldFile = oldFiles[file.name];
+                        return {
+                            ...oldFiles,
+                            [file.name]: { ...oldFile, progress },
+                        };
+                    });
+                }, abort.signal);
```

Comment:
You may also like to specify `"tag": "-"` here if you didn't want to replace existing files....

Reply:
Where do I pass it? Do I pass it to options?

Reply:
From matrix => "i think this is a "version 2" thing :)"